Microsoft.Toolkit.Uwp.UI.Controls/QuickReturnHeader/QuickReturnHeader.cs

```diff
@@ -0,0 +1,248 @@
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+using System;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Hosting;
+using Windows.UI.Xaml.Media;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Quick return header control to be used with ListViews
+    /// </summary>
+    public class QuickReturnHeader : ContentControl
+    {
+        private ScrollViewer _scrollViewer;
+        private double _previousVerticalScrollOffset;
+        private CompositionPropertySet _scrollProperties;
+        private CompositionPropertySet _animationProperties;
+        private Visual _headerVisual;
+
+        public QuickReturnHeader()
+        {
+            HorizontalContentAlignment = HorizontalAlignment.Stretch;
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="IsQuickReturnEnabled"/> property.
+        /// </summary>
+        public static readonly DependencyProperty IsQuickReturnEnabledProperty =
+            DependencyProperty.Register(nameof(IsQuickReturnEnabled), typeof(bool), typeof(QuickReturnHeader), new PropertyMetadata(true, OnIsQuickReturnEnabledChanged));
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the quick return header is enabled.
+        /// If true the quick return behavior is used.
+        /// If false regular header behavior is used.
+        /// Default is true.
+        /// </summary>
+        public bool IsQuickReturnEnabled
+        {
+            get { return (bool)GetValue(IsQuickReturnEnabledProperty); }
+            set { SetValue(IsQuickReturnEnabledProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="IsSticky"/> property.
+        /// </summary>
+        public static readonly DependencyProperty IsStickyProperty =
+            DependencyProperty.Register(nameof(IsSticky), typeof(bool), typeof(QuickReturnHeader), new PropertyMetadata(false, OnIsStickyChanged));
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the quick return header should always be visible.
+        /// If true the header is always visible.
+        /// If false the header will move out of view when scrolling down.
+        /// Default is false.
+        /// </summary>
+        public bool IsSticky
+        {
+            get { return (bool)GetValue(IsStickyProperty); }
+            set { SetValue(IsStickyProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the ListView this header belongs to
+        /// </summary>
+        public ListView TargetListView { get; set; }
```

Comment:
@pdehne you should be able to use FindAscentant<ListViewBase> and get the ListView (or GridView) https://github.com/Microsoft/UWPCommunityToolkit/blob/dev/Microsoft.Toolkit.Uwp.UI/Extensions/VisualTreeExtensions.cs#L93

In theory, this should also work for custom controls, but if it doesn;t you still have the TargetListView D.P. to fallback on.